File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/data/cache/TimeCache.java
package org.kevoree.modeling.api.data.cache;

import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.api.data.CacheEntry;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by duke on 10/30/14.
 */
public class TimeCache {
    public Map<Long, CacheEntry> payload_cache = new HashMap<Long, CacheEntry>();
    public KObject root = null;
    public boolean rootDirty = false;
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/data/cache/TimeCache.ts
///<reference path="../../KObject.ts"/>
///<reference path="../../ModelListener.ts"/>
///<reference path="../../../../../../java/util/JUArrayList.ts"/>
///<reference path="../../../../../../java/util/JUHashMap.ts"/>
///<reference path="../../../../../../java/util/JUList.ts"/>
///<reference path="../../../../../../java/util/JUMap.ts"/>

class TimeCache {

  public obj_cache: JUMap<number, KObject<any,any>> = new JUHashMap<number, KObject<any,any>>();
  public payload_cache: JUMap<number, any[]> = new JUHashMap<number, any[]>();
  public root: KObject<any,any> = null;
  public rootDirty: boolean = false;
  public listeners: JUList<ModelListener> = new JUArrayList<ModelListener>();
  public obj_listeners: JUMap<number, JUList<ModelListener>> = new JUHashMap<number, JUList<ModelListener>>();

}

